### DISCUSSION TITLE
Make Grapes identify childrens in react components

### DISCUSSION THREAD

**Discussion by julia-alberici** [GrapesJS/grapesjs#4955] [Q&A]
Hi! Here's my situation: I have a React component, something like this 
<img src="https://user-images.githubusercontent.com/101437173/222434373-034ce111-c560-4aa8-8a01-e1e6dde9da63.png" width="50%" />
When a define a grapes component using plane HTML (the same HTML that a have in my React component) will result in this: 
Code             |  Result (component with 3 components inside)
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/101437173/222436433-afaf18b4-6d0b-471b-93d1-610fd629fab1.png)  | ![image](https://user-images.githubusercontent.com/101437173/222436444-538f741d-468f-4842-9b4f-c898056915ad.png) 

But defining it using my component as content will result in just one component without the childrens 

Code             |  Result (component without components inside)
:-------------------------:|:-------------------------:
![image](https://user-images.githubusercontent.com/101437173/222438443-c369644c-bbe0-4ca0-b6e8-b99eeac3b825.png)  | ![image](https://user-images.githubusercontent.com/101437173/222438866-3b802524-5997-4394-929f-06566061d9d6.png)

My question is: there's a way of defining the block content as react element and still have grapes recognizing that the elements inside of it are components?
I aprecciate any help in advance. Here's the sandbox that i used to exemplify my problem https://codesandbox.io/s/grapesjs-react-components-forked-o34oc8?file=/src/react-components.js

**Comment by artf** [comment]
> there's a way of defining the block content as react element and still have grapes recognizing that the elements inside of it are components?

The concept is the same as when you use a React component from another external library in your app, you can't access its internal stuff, you can only rely on what the developer has exposed for you.
GrapesJS can't access its internal code but only update its behaviour by using its exposed API/props.

The point here is, it's up to you to understand how you to split your components and how customizable you want to make for your users.

**Reply by julia-alberici** [reply]
Heyy @artf, thank you for taking a time to answer me. I understand what you said but in this case i really can use some help here, let me be more specific about my situation. I have a input that comes from a external library (it's "mine" lib so i can add or rmv anything needed for this btw) and this input has 2 props that are "prefix" and "suffix" both receive a component 
![image](https://user-images.githubusercontent.com/101437173/222825611-8edd419c-ba24-4f53-a3f4-333efbef79bd.png)
my point is: i want to make props "prefix" and "suffix" customizable allowing users to drop grapes components in there. Do you have any idea or guidence to provide me that allow me to accomplish this task without having to expose the entire component?

**Reply by artf** [reply]
Yeah sure this is a valid approach passing components via props but I what is missing in the GrapesJS react component renderer is the possibility to pass and render other components via props (now it only renders its children).
